### PR TITLE
14 merchant record endpoint

### LINKED ISSUE #14: Merchant record endpoint
INDEX: GET /api/v1/merchants.json returns all records as an array of JSON objects

SHOW: GET /api/v1/merchants/1.json returns one JSON object

FIND: GET /api/v1/merchants/find?parameters returns one record as a JSON object based on the params (e.g. GET /api/v1/merchants/find?name=Schroeder-Jerde)

FIND_ALL: GET /api/v1/merchants/find_all?parameters returns all records as an array of JSON objects (e.g. GET /api/v1/merchants/find_all?name=Cummings-Thiel) - even if there is only one returned value, it should still return an array

RANDOM: api/v1/merchants/random.json returns one JSON object

### PR DESCRIPTION
**Pull request by kheppenstall**
-Change merchant name datatype from string to citext (2 migrations)
-Add serializer gem and merchant serializer
-Add a Merchants Controller within modules Api and V1
-Create 'api/v1/merchants' endpoint with an index method
-Create 'api/v1/merchants/:id' endpoint with a show method

Closes #14

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -44,9 +44,9 @@
   end
 
   create_table "merchants", force: :cascade do |t|
-    t.string   "name"
     t.datetime "created_at", null: false
     t.datetime "updated_at", null: false
+    t.citext   "name"
```
Comment (ski-climb) [inline db/schema.rb:49]: Great find with citext!  Thanks for pulling that in.

**Comment by kheppenstall** [comment]
@sallymacnicholas We are going to merge this in but would still love some feedback on this bit of code since it is our first time making an API.

File: spec/requests/api/v1/endpoints/merchants_spec.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    merchant1, merchant2, merchant3 = create_list(:merchant, 3)
+    name = merchant1.name
+
+    get '/api/v1/merchants'
+    merchants = JSON.parse(response.body)
+    merchant = merchants.first
+
+    expect(response).to be_success
+    expect(merchants.count).to eq 3
+    expect(merchant['name']).to eq name
```
Comment (ski-climb) [inline spec/requests/api/v1/endpoints/merchants_spec.rb:14]: Excellent - we're checking to make sure we get merchants back in the index without going overboard.

File: spec/requests/api/v1/endpoints/merchants_spec.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    create(:merchant, id: 5)
+
+    get '/api/v1/merchants/5'
+    merchant = JSON.parse(response.body)
+
+    expect(response).to be_success
+
+    expect(merchant).to be_a Hash
+    expect(merchant.keys.count).to eq 2
+    expect(merchant).to have_key 'id'
```
Comment (ski-climb) [inline spec/requests/api/v1/endpoints/merchants_spec.rb:27]: I like how you're more specific for the show action.  Good call.

**Review by ski-climb — APPROVED** [review]

**Pull request merged**